docs/backend-system/core-services/01-index.md

```diff
@@ -65,10 +88,366 @@ createBackendPlugin({
+        const urls = await Promise.all[
+          discovery.getBaseUrl('derp'),
+          discovery.getExternalBaseUrl('derp'),
+        ];
+
+        await Promise.all(
+          urls.map(
+            (url) => fetch(url).then((r) => r.json()),
+          ),
+        );
+      },
+    });
+  },
+});
+```
+
+## Identity
+
+When working with backend plugins, you might find that you will need to interact with the `auth-backend` plugin to both authenticate backstage tokens, and get things like the `entityRef` of the authenticated user, and anything that they might claim to own through `ownershipEntityRefs`.
+
+### Using the service
+
+The following example shows how to get the `IdentityService` in your `example` backend plugin and retrieve the users `entityRef` and ownership claims for the incoming `http` request.
+
+```ts
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { Router } from 'express';
+
+createBackendPlugin({
+  id: 'example',
+  register(env) {
+    env.registerInit({
+      deps: {
+        identity: coreServices.identity,
+        http: coreServices.httpRouter,
+      },
+      async init({ http, identity }) {
+        const router = Router();
+        router.get('/test-me', (request, response) => {
+          // use the identityService pull out the header from the request and get the user
+          const {
+            identity: { userEntityRef, ownershipEntityRefs },
+          } = await identity.getIdentity({
+            request,
+          });
+
+          // sent the decoded and validated things back to the user
+          response.json({
+            userEntityRef,
+            ownershipEntityRefs,
+          });
+        });
+
+        http.use(router);
+      },
+    });
+  },
+});
+```
+
+### Configuration of the service
+
+There's additional configuration that you can optionally pass to setup the `identity` core service.
+
+- `issuer` - Set an optional issuer for validation of the `jwt`
+- `algorithms` - `jws` `alg` header for validation of the `jwt`, defaults to `ES256`. More info on supported algorithms under [jose](https://github.com/panva/jose)
+
+You can configure these additional options by adding an override for the core service when calling `createBackend` like follows:
+
+```ts
+import { identityFactory } from '@backstage/backend-app-api`;
+
+const backend = createBackend({
+  services: [
+    identityFactory({
+      issuer: 'backstage',
+      algorithms: ['ES256', 'RS256']
+    }),
+  ],
+});
+```
+
+## Lifecycle
+
+When writing plugins, it's often that you will have long running things that you might want to ensure clean shutdowns of when the plugins are torn down, or when the backend is quit (think local development). You shouldn't have to worry too much about providing shutdowns for any of the core services that you use, and should really only need to take care of anything that you create that you should stop when your plugin stops.
```

Comment:
```suggestion
This service allows your plugins to register hooks for cleaning up resources as the service is shutting down (e.g. when a pod is being torn down, or when pressing `Ctrl+C` during local development). Other core services also leverage this same mechanism internally to stop themselves cleanly.
```

Reply:
Wasn't sure about the original core services formulation 🤔 Does this one make sense?